python_modules/dagster/dagster/core/definitions/decorators/config_mapping.py

```diff
@@ -0,0 +1,76 @@
+from typing import Any, Callable, Optional, Union
+
+from dagster import check
+from ..config import ConfigMapping
+
+def config_mapping(
+    config_fn: Callable[..., Any] = None,
+    config_schema: Optional[Any] = None,
+    receive_processed_config_values: Optional[bool] = None,
+) -> Union[_ConfigMapping, ConfigMapping]:
+    """Create a config mapping with the specified parameters from the decorated function.
+
+    This shortcut simplifies the core :class:`ConfigMapping` API by omitting additional
+    parameters when they are not needed. The config schema will be inferred from the type
+    signature of the decorated function if not explicitly provided.
+
+    Args:
+        config_schema (ConfigSchema): The schema of the composite config.
+        receive_processed_config_values (Optional[bool]): If true, config values provided to the config_fn
+            will be converted to their dagster types before being passed in. For example, if this
+            value is true, enum config passed to config_fn will be actual enums, while if false,
+            then enum config passed to config_fn will be strings.
+
+
+    Examples:
+
+        .. code-block:: python
+
+            @op
+            def my_op(context):
+                return context.op_config["foo"]
+
+            @graph
+            def my_graph():
+                my_op()
+
+            @config_mapping
+            def my_config_mapping(val):
+                return {"ops": {"my_op": {"config": {"foo": val["foo"]}}}}
+            
+            @config_mapping(config_schema={"foo": str})
+            def my_config_mapping(val):
+                return {"ops": {"my_op": {"config": {"foo": val["foo"]}}}}
+
+    """
+    # This case is for when decorator is used bare, without arguments. e.g. @config_mapping versus @config_mapping()
+    if callable(config_fn):
+        check.invariant(config_schema is None)
+        check.invariant(receive_processed_config_values is None)
+
+        return _ConfigMapping()(config_fn)
+
+    check.invariant(config_fn is None)
+    return _ConfigMapping(
+        config_schema=config_schema,
+        receive_processed_config_values=receive_processed_config_values,
+    )
+
+class _ConfigMapping:
+    def __init__(
+        self,
+        config_schema: Optional[Any] = None,
+        receive_processed_config_values: Optional[bool] = None,
+    ):
+        self.config_schema = config_schema
+        self.receive_processed_config_values = check.opt_bool_param(receive_processed_config_values, "receive_processed_config_values")
+
+    def __call__(self, fn: Callable[..., Any]) -> ConfigMapping:
+        check.callable_param(fn, "fn")
+
+        config_mapping_def = ConfigMapping(
+            config_fn=fn,
+            config_schema=self.config_schema,
+            receive_processed_config_values=self.receive_processed_config_values,
+        )
+        return config_mapping_def
```

Comment:
nit: `config_mapping_def` name doesn't really make sense, since it isn't a definition object.